libcontainer/process_linux.go

```diff
@@ -580,7 +580,18 @@ func (p *initProcess) start() (retErr error) {
 	// cgroup. We don't need to worry about not doing this and not being root
 	// because we'd be using the rootless cgroup manager in that case.
 	if err := p.manager.Apply(p.pid()); err != nil {
-		return fmt.Errorf("unable to apply cgroup configuration: %w", err)
+		if errors.Is(err, cgroups.ErrRootless) {
+			// ErrRootless is to be ignored except when the
+			// container doesn't have private pidns.
+			if !p.config.Config.Namespaces.IsPrivate(configs.NEWPID) {
```

Comment:
Any reason to not put this condition in the previous if? Do you think it is more readable like this?

Reply:
I did that initially, but rolled it back later since changing the code like this complicates the review (the whole block changes instead of just one line).

I will add a separate commit that does it.

Reply:
My bad; I mixed this up with something else. Updated; PTAL @rata

Reply:
Ughm, I just broke everything.

This is two `if` statements (and an `else`) because we want to
 - ignore ErrRootless;
 - except there's no private pidns, in which case we want a warning;
 - return all other errors as is.
 
 No way to do that with a single `if`.

Reply:
Makes sense, I tried to collapse it and I'm not sure it is more readable than this.

Reply:
It's not a question of readability. There are _three_ different conditions, can't do it with a single if.